src/NServiceBus.Core/EndpointInstance.cs

```diff
@@ -1,21 +1,33 @@
 ﻿namespace NServiceBus
 {
-    using System;
-    using System.Text;
+    using System.Collections.Generic;
+    using System.Linq;
 
     /// <summary>
     /// Represents a name of an endpoint instance.
     /// </summary>
     public sealed class EndpointInstance
     {
+        readonly PropertyDictionary properties;
+        
+        /// <summary>
+        /// Returns the name of the endpoint.
+        /// </summary>
+        public Endpoint Endpoint { get; }
+
+        /// <summary>
+        /// A specific discriminator for scale-out purposes.
+        /// </summary>
+        public string ScaleOutDiscriminator { get; }
```

Comment:
Not sure about committing to make satellites fully supported concepts. I am on the fence here. On the other hand, no matter how the current implementation of the satellites is broken, the concept itself makes sense and I see there will be need for privately-scoped "sub addresses"...